Sources/Subs.php

```diff
@@ -6520,11 +6520,17 @@ function set_tld_regex($update = false)
 function build_regex($strings, $delim = null, $returnArray = false)
 {
 	global $smcFunc;
+	static $cache_buids = array();
 
 	// If it's not an array, there's not much to do. ;)
 	if (!is_array($strings))
 		return preg_quote(@strval($strings), $delim);
 
+	$cache_key = md5(json_encode($strings)) .'d=' . $delim . 'r=' . $returnArray;
```

Comment:
Please change `$cache_key` to `$regex_key` here and in the subsequent code.

Reply:
Please change 
```
md5(json_encode($strings)) .'d=' . $delim . 'r=' . $returnArray; 
```
to 
```
md5(json_encode($strings) . json_encode(array($delim, $returnArray)); 
```
This will be more reliable.